app/src/main/kotlin/com/intive/picover/common/viewmodel/MVIViewModel.kt

```diff
@@ -1,51 +1,26 @@
 package com.intive.picover.common.viewmodel
 
-import androidx.compose.runtime.MutableState
-import androidx.compose.runtime.State
-import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.intive.picover.common.viewmodel.event.Event
-import com.intive.picover.common.viewmodel.sideeffect.SideEffect
-import com.intive.picover.common.viewmodel.state.MVIState
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 
-abstract class MVIViewModel<S : MVIState, E : Event, SE : SideEffect>(initialState: S) : ViewModel() {
+abstract class MVIViewModel<S : Any, E : Any, SE : Any>(initialState: S) : ViewModel() {
 
 	@Suppress("ktlint:standard:property-naming")
-	protected val _state = mutableStateOf(initialState)
-	val state: State<S> = _state
+	protected val _state = MutableStateFlow(initialState)
+	val state = _state.asStateFlow()
```

Comment:
One advantage of state flow is that it's easy to test the stream of state changes with turbine. I'm not sure if that's so easy with compose State. It would be also nice to decouple view model logic from compose related classes

Reply:
Decupling seems to be most important here :)